### ISSUE TITLE
loc/ids for symbols and sentences

### ISSUE THREAD

**Issue by tillmo** [ontohub/ontohub#1585]
Symbols and sentences can have the same name. How should we differentiate their loc/ids? There are the following options:
1. forbid symbols and sentences to have the same name
2. rename sentences if their name conflicts with a symbol name
3. add markers (e.g. `sym` and `sen`) in the loc/id path to distinguish between symbols and sentences

**Comment by tillmo** [comment]
I tend to option 2. After all, sentence names are usually not referenced from outside, and therefore their renaming is not harmful.

**Comment by tillmo** [comment]
@mcodescu, what do you think?

**Comment by eugenk** [comment]
Wouldn't option 2 introduce problems with views and imports? Of course Hets and Ontohub would not encounter any technical problems because the actual sentences are not modified, but the Ontohub user might get confused.

**Comment by eugenk** [comment]
Oh, we use the sentence name for sending the list of axioms/the theorem to Hets when proving. If we decide to rename sentences, we need to add another attribute `original_name` that we use for communication with hets (e.g. for proving).

**Comment by tillmo** [comment]
If Ontohub does the renaming already when parsing ontologies, Ontohub users shouldn't be confused. But you are right that Hets needs to original name for proving. So altogether, option 1 might be better.

**Comment by tillmo** [comment]
https://github.com/spechub/Hets/pull/1630 is at least a partial fix. Does it suffice (see example and discussion there)?

**Comment by eugenk** [comment]
Yes, this should suffice. We still need to catch the error and present it to the user. Currently, we get with the version of spechub/Hets#1630:
```
PG::CheckViolation: ERROR:  new row for relation "ontologies" violates check constraint "logic_id_check"
DETAIL:  Failing row contains (24, SingleOntology, null, null, null, 24, done, Cat2, null, f, 0, 0, 1, 0, 0, 2016-05-09 07:02:51.691066, 2016-05-09 07:02:54.438793, 2, null, null, null, null, null, null, cat2, .clif, t, t, 0, 0).
: UPDATE "ontologies" SET "state" = 'done', "updated_at" = '2016-05-09 07:02:54.438793' WHERE "ontologies"."type" IN ('SingleOntology') AND "ontologies"."id" = 24
Download
```

**Comment by tillmo** [comment]
spechub/Hets#1630 has been merged now.

**Comment by eugenk** [comment]
This has been solved by #1618.
